src/main/java/gregicadditions/machines/multi/miner/TileEntitySteamMiner.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    @Override
+    public void update() {
+        super.update();
+        if (!getWorld().isRemote) {
+            fillInternalTankFromFluidContainer(fluidContainerInventory, fluidContainerInventory, 0, 1);
+            // if sufficient amounts of steam and drilling fluid aren't present, do nothing
+            if ((importFluids.getTankAt(0).getFluidAmount() < 1) || (importFluids.getTankAt(1).getFluidAmount() < 16)) {
+                return;
+            } else {
+                System.out.println("im draining brooooooo");
+            }
+
+            importFluids.getTankAt(0).drain(minerType.drillingFluidConsumePerTick, true);
+            importFluids.getTankAt(1).drain(16, true);
+
+            WorldServer world = (WorldServer) this.getWorld();
+            Chunk chuck = world.getChunk(getPos());
+            ChunkPos chunkPos = chuck.getPos();
+            if (x.get() == Long.MAX_VALUE || x.get() == 0) {
+                x.set(chunkPos.getXStart());
+            }
+            if (z.get() == Long.MAX_VALUE || z.get() == 0) {
+                z.set(chunkPos.getZStart());
+            }
+            if (y.get() == Long.MAX_VALUE || y.get() == 0) {
+                y.set(getPos().getY());
+            }
+
+            List<BlockPos> blockPos = Miner.getBlockToMinePerChunk(this, x, y, z, chuck.getPos());
+            blockPos.forEach(blockPos1 -> {
+                NonNullList<ItemStack> itemStacks = NonNullList.create();
+                IBlockState blockState = this.getWorld().getBlockState(blockPos1);
+                blockState.getBlock().getDrops(itemStacks, world, blockPos1, blockState, 0);
+                if (addItemsToItemHandler(exportItems, true, itemStacks)) {
+                    addItemsToItemHandler(exportItems, false, itemStacks);
+                    world.destroyBlock(blockPos1, false);
+                }
+            });
+
+            if (!getWorld().isRemote && getTimer() % 5 == 0) {
+                pushItemsIntoNearbyHandlers(getFrontFacing());
+            }
+        }
+    }
+
+    @Override
+    public void addInformation(ItemStack stack, @Nullable World player, List<String> tooltip, boolean advanced) {
+        tooltip.add(I18n.format("gtadditions.machine.miner.steam.description"));
+        tooltip.add(I18n.format("gtadditions.machine.miner.fluid_usage", minerType.drillingFluidConsumePerTick, I18n.format(Materials.DrillingFluid.getFluid(0).getUnlocalizedName())));
+    }
+
+    @Override
+    public NBTTagCompound writeToNBT(NBTTagCompound data) {
+        super.writeToNBT(data);
+        data.setTag("xPos", new NBTTagLong(x.get()));
+        data.setTag("yPos", new NBTTagLong(y.get()));
+        data.setTag("zPos", new NBTTagLong(z.get()));
+        return data;
+    }
+
+    @Override
+    public void readFromNBT(NBTTagCompound data) {
+        super.readFromNBT(data);
+        x.set(data.getLong("xPos"));
+        y.set(data.getLong("yPos"));
+        z.set(data.getLong("zPos"));
+    }
+
```

Comment:
The items in the Fluid Container slots are voided if the machine is broken. This section of code will fix that (you will have to import `net.minecraft.block.Block`):
```suggestion

        @Override
        public void onRemoval() {
            for (int slot = 0; slot < fluidContainerInventory.getSlots(); slot++)
                Block.spawnAsEntity(getWorld(), getPos(), fluidContainerInventory.getStackInSlot(slot));
        }

```

Reply:
Good catch, that's been added now.